numba/tests/test_typedlist.py

```diff
@@ -155,6 +155,26 @@ def test_unsigned_access(self):
         self.assertEqual(L.pop(ui32_1), 2)
         self.assertEqual(L.pop(ui32_0), 123)
 
+    def test_dtype(self):
+        L = List()
+        self.assertEqual(L._dtype, None)
+
+        L = List.empty_list(int32)
+        self.assertEqual(L._dtype, int32)
+
+        L = List.empty_list(float32)
+        self.assertEqual(L._dtype, float32)
+
+        @njit
+        def foo():
+            li, lf = List(), List()
+            li.append(int32(1))
+            lf.append(float32(1.0))
+            return li._dtype, lf._dtype
+
+        self.assertEqual(foo(), (np.dtype('int32'), np.dtype('float32')))
+        self.assertEqual(foo.py_func(), (int32, float32))
```

Comment:
I did this since the jitted function returns numpy dtypes whereas the py_func returns numba types.

Reply:
Yes, this is the current design and it is intended to work like this. If this is a good idea or not is another matter.